routes/routes.ts

```diff
@@ -284,6 +284,55 @@ router.post("/category", async (req: Request, res: Response) => {
   }
 });
 // route to update a category (including adding a bookmark to it)
+router.patch("/category/:id", async (req: Request, res: Response) => {
+  try {
+    // user has to have a session
+    if (req.session.userId) {
+      const userId = req.session.userId;
+      const categoryId = req.params.id;
+      const categoryRef = usersRef
+        .child(userId)
+        .child("categories")
+        .child(categoryId);
+      await categoryRef.update(
```

Comment:
I think we have several ways of updating a category;
- Adding a bookmark to it
- Deleting a bookmark from it
- Changing individual attributes like the category name, description, etc.

We need a way to specify what type of update the client is trying to make, then use that type to determine what gets updated in the firebase doc.

We could use this route for handling updates(add, delete) to the categories bookmarks. Maybe the request body could have a property with two possible values of add or delete, then we first fetch the doc for that category, append to it and update the doc if we are adding a new bookmark or remove a bookmark with the provided bookmark id from our request body. This means each bookmark should have an id, if it already doesn't have.

Reply:
We can create a separate route for updating the category attributes(name, description, etc).

Reply:
> I think we have several ways of updating a category;
> 
> * Adding a bookmark to it
> * Deleting a bookmark from it
> * Changing individual attributes like the category name, description, etc.
> 
> We need a way to specify what type of update the client is trying to make, then use that type to determine what gets updated in the firebase doc.
> 
> We could use this route for handling updates(add, delete) to the categories bookmarks. Maybe the request body could have a property with two possible values of add or delete, then we first fetch the doc for that category, append to it and update the doc if we are adding a new bookmark or remove a bookmark with the provided bookmark id from our request body. This means each bookmark should have an id, if it already doesn't have.

This makes sense. the req body can have a 'update_type' field (which could be an enum with two possible values), then based on that field, I'll update the category accordingly. I'll get on it now. Thanks for your help.

Reply:
> We can create a separate route for updating the category attributes(name, description, etc).

alright. will do.

Reply:
Yes, this works!